libraries/ms-common/inc/pwm_input.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+// PWM Input Module
+
+#include <stdint.h>
+
+#include "pwm_mcu.h"
+#include "status.h"
+
+typedef void (*PwmInputHandler)(uint32_t dc, uint32_t period, void *context);
+
+// Don't worry about initializing this struct. Can just pass in an empty one
+// as long as memory has been allocated for it. The PWM driver will use it as storage
+typedef struct {
+  PwmInputHandler callback;
+  void *context;
+} PwmInputStorage;
+
+//
+// Settings struct
+//
+
+// This settings struct can have a callback which will be called
+// whenever a new PWM reading is read UNLESS the DC value is 0. This
+// is because the timer interrupt does not trigger for values of 0 (there is no rising edge).
+// As a result, it is advised that the user also calls |pwm_input_get_dc| sometimes as this
+// will force the callback to fire for DC values of 0.
+
+// As well, there is a channel parameter. It can only be a value of one or two. This is
+// because the |TIM_PWMIConfig| only works on those two channels.
+typedef struct {
+  PwmInputHandler callback;
+  void *context;
+  PwmChannel channel;
+} PwmInputSettings;
+
+// Initializes the timer for PWM input. In order to choose a timer, and a timer channel
```

Comment:
Ok, that makes sense. It's not actually possible for the user to select `TIM2` (it's not in the `PwmTimers` struct. I'll add a comment to let the user know they should be careful. Yes, collisions would be a problem. This driver won't play nice with PWM output at the moment, since PWM output works on all channels. Will warn the hardware team about that as well

Reply:
Made a comment on https://uwmidsun.atlassian.net/wiki/spaces/ELEC/pages/27131906/Rules+Guidelines

Hopefully that's enough to prevent conflicts